lib/innit-client/src/lib.rs

```diff
@@ -85,18 +85,43 @@ impl InnitClient {
     }
 
     pub async fn get_parameter(&self, name: String) -> Result<GetParameterResponse> {
-        let url = self.base_url.join("parameter")?.join(&name)?;
+        let url = self.join_path("parameter", &name)?;
         let resp = self.client.get(url).send().await?.error_for_status()?;
         let parameter = resp.json::<GetParameterResponse>().await?;
 
         Ok(parameter)
     }
 
     pub async fn get_parameters_by_path(&self, path: String) -> Result<ListParametersResponse> {
-        let url = self.base_url.join("parameters")?.join(&path)?;
+        let url = self.join_path("parameters", &path)?;
         let resp = self.client.get(url).send().await?.error_for_status()?;
         let parameters = resp.json::<ListParametersResponse>().await?;
 
         Ok(parameters)
     }
+
+    fn join_path(&self, base_segment: &str, path: &str) -> Result<Url> {
```

Comment:
This is really stupid. Subsequent `.join` calls were smashing over each other due to `/` placement in the path, so this gets around that